crates/sui-types/src/zk_login_authenticator.rs

```diff
@@ -202,15 +109,18 @@ impl AuthenticatorTrait for ZkLoginAuthenticator {
                 error: "User signature verify failed".to_string(),
             });
         }
-
-        // Finally, verify the Groth16 proof against public inputs and proof points.
-        // Verifying key is pinned in fastcrypto.
-        match verify_zk_login_proof_with_fixed_vk(&self.proof, &self.public_inputs) {
-            Ok(true) => Ok(()),
-            Ok(false) | Err(_) => Err(SuiError::InvalidSignature {
-                error: "Groth16 proof verify failed".to_string(),
-            }),
-        }
+        let mut extended_pk_bytes = vec![self.user_signature.scheme().flag()];
+        extended_pk_bytes.extend(self.user_signature.public_key_bytes());
+        verify_zk_login(
```

Comment:
based on the new circuit design, other than the epoch check that is done in sui, everything is delegated to verify_zk_login and maintained in fastcrypto